### DISCUSSION TITLE
Complete sampling overhaul

### DISCUSSION THREAD

**Discussion by darenager** [SynthstromAudible/DelugeFirmware#891] [Ideas]
Sampling on the Deluge, then saving the sample is frankly the most unintuitive of any sampler I have used in 30 years, it is needlessly convoluted and lacking in basic functionality - such as threshold recording, trimming and saving, sampling for a specified duration etc.

Any enhancements to address these shortcomings would be most welcome and I think make the Deluge a bit more user friendly and appealing to a wider audience.

I started this discussion to see if others have ideas how this could be improved and indeed if is as much an issue for other users as it is for me.

**Comment by jamiefaye** [comment]
I would like to see a "Track Freeze" feature (like Ableton). This will help us cope with processing overhead as one can "bounce" a kit or synth track to a .WAV equivalent while keeping the underlying note events and automations around.

**Reply by darenager** [reply]
Excellent idea!

**Reply by trappar** [reply]
Also see: https://github.com/SynthstromAudible/DelugeFirmware/discussions/263

**Comment by darenager** [comment]
Threshold recording would be perhaps one of the handiest for me, for drum kit creation especially.
